storage/sis-storage/src/main/java/org/apache/sis/storage/tilematrix/TileMatrixSet.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sis.storage.tilematrix;
+
+import java.util.Collection;
+import java.util.Optional;
+import org.opengis.geometry.Envelope;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.util.GenericName;
+
+/**
+ * A TileMatrixSet is a collection of {@linkplain TileMatrix} in the same CRS but at different
+ * scale levels.
+ *
+ * <p>
+ * If the {@linkplain CoordinateReferenceSystem } of the TileMatrixSet has more
+ * then two dimensions, it is possible to find {@linkplain TileMatrix} at the same scale.
+ * Each TileMatrix been located on a different slice in one of the {@linkplain CoordinateReferenceSystem }
+ * axis.
+ * </p>
```

Comment:
This sentence is true only in some particular cases, when the CRS axes are colinear with grid axes. This is often the case but not necessarily (e.g. rotation, or some map projections). More generally, slices are located at different points on a _grid_ axis (not a CRS axis).